src/xtt.c

```diff
@@ -106,7 +106,8 @@ enftun_xtt_handshake(const char** server_hosts,
                      const char* basename_in,
                      struct enftun_xtt* xtt)
 {
-    struct enftun_tcp sock;
+    struct enftun_tcp_native sock;
```

Comment:
Rename `sock` to `sock_native`.

Add a `struct enftun_tcp* sock`, and set `sock = &sock_native.base`.